packages/mapviewer/src/modules/map/components/CompareSlider.vue

```diff
@@ -36,20 +36,22 @@ watch(
         if (oldLayerOnTop) {
             unRegisterRenderingEvents(oldLayerOnTop.id, oldLayerOnTop.uuid)
         }
-
         if (getLayerFromMapById(newLayerOnTop.id, newLayerOnTop.uuid)) {
             registerRenderingEvents(newLayerOnTop.id, newLayerOnTop.uuid)
             olMap.render()
         } else {
-            // The layer config is always modified before the map, which means the
-            // visible layer on top according to the config could not exist within
-            // the map. This means that if we have layers that loads after the component is
-            // instantiated, most often COG layers.
+            // There are cases where the layer config and layer store are
+            // modified and updated before the map. This means we need to delay
+            // the event which bind the rendering event to the layer. This
+            // happens in general at startup, especially if the layer itself is
+            // quite big.
 
-            // to mitigate the issue, we ask the map to register the events once after the first render,
-            // after all layers have been loaded, and re-render it.
-            // We should arrive here ONLY if we arrive on startup on a map with an active compare slider.
-            olMap?.once('rendercomplete', () => {
+            // To mitigate the issue, we ask the map add the rendering functions
+            // either when the 'precompose' event is fired (for WebGL layers) or
+            // a 'rendercomplete' event is fired (for canvas layers).
+            // This ensure we link the rendering function to the layer as soon
+            // as possible.git
+            olMap?.once(shouldUseWebGlContext.value ? 'precompose' : 'rendercomplete', () => {
```

Comment:
I'd maybe leave both event.
Are you 100% sure `precompose` is only used in the WebGL context?

Reply:
I'm 100% sure (at least according to the documentation.)